src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -514,11 +514,49 @@ internal static IntroducedMethodEnumerator GetIntroducedMethods(RuntimeType type
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern void GetNextIntroducedMethod(ref RuntimeMethodHandleInternal method);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern bool GetFields(RuntimeType type, IntPtr* result, int* count);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "RuntimeTypeHandle_GetFields")]
+        private static partial Interop.BOOL GetFields(MethodTable* pMT, Span<IntPtr> data, ref int usedCount);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern Type[]? GetInterfaces(RuntimeType type);
+        internal static bool GetFields(RuntimeType type, ref Span<IntPtr> buffer, ref int count)
+        {
+            TypeHandle typeHandle = type.GetNativeTypeHandle();
+
+            CorElementType elementType = (CorElementType)typeHandle.GetCorElementType();
+            if (elementType is CorElementType.ELEMENT_TYPE_VAR or CorElementType.ELEMENT_TYPE_MVAR)
+            {
+                throw new ArgumentException(SR.Arg_InvalidHandle);
+            }
```

Comment:
```suggestion
            Debug.Assert(!IsGenericVariable(type));
            
```
Same as `GetInterfaces`?

Reply:
I can, but I don't see the same protections in the `GetFields` callers. I'll change it and see what falls out.